### ISSUE TITLE
personal config ?

### ISSUE THREAD

**Issue by biocyberman** [chrisbarrett/spacemacs-layers#13]
Firstly, thanks for sharing the repo. It teaches me many things about how to config spacemacs in a very advanced way. 
I created a `~/.emacs.d/private/custom.el` and modify the code in `cb-bootstrap` like so:

```
(defun cb-bootstrap/user-config ()
  (setq custom-file (concat user-emacs-directory "private/custom.el"))
  (when (file-exists-p custom-file)
    (load custom-file))
  (with-demoted-errors "Personal config: %S"
    (require 'personal-config nil t)))

```

In the `custom.el` i have file contents like so:

```
;;; custom.el --- Settings to be loaded inside `dotspacemacs/user-config' -*- lexical-binding: t; -*-

;;; Commentary:

;;; Code:

;; For search engines
(setq browse-url-browser-function 'browse-url-generic
      engine/browser-function 'browse-url-generic
      browse-url-generic-program "google-chrome")
(add-hook 'doc-view-mode-hook 'auto-revert-mode)
(add-hook 'python-mode-hook 'outline-minor-mode)

(provide 'personal-config)

;;; custom.el ends here

```

I restarted emacs, reloaded dot files with `SPC f e d`
However, when I do `SPC f e p` I get this error:

`find-library-name: Can't find library personal-config`

I can edit the keybindings so it open/find the file directly. However, I want to have your advice about this.

**Comment by chrisbarrett** [comment]
TLDR: You should
1. revert your changes to cb-bootstrap.el
2. `mv custom.el personal-config.el`
3. Prepend `(add-to-list 'load-path "~/.emacs.d/private")` to the contents of `dotspacemacs/user-config` in ~/.spacemacs

That should automatically get your personal config loaded.
----
The file pointed to by the `custom-file` variable is managed by Emacs. Emacs will automatically write to it when you modify variables using the customisation features under `M-x customize`. You shouldn't edit that file directly unless you know how that all works. :)

My bootstrap process searches for a file called `personal-config.el` on the lisp load-path. `~/.emacs.d/private` probably isn't on the load path, so you'll need to add it: 

```elisp
(defun dotspacemacs/user-config ()
  "Configuration function for user code.
This function is called at the very end of Spacemacs initialization after
layers configuration. You are free to put any user code."
  (setq powerline-default-separator 'utf-8)
  (add-to-list 'load-path "~/.emacs.d/private") # <---- Add this line before the call to cb-bootstrap/user-config
  (cb-bootstrap/user-config)
  (cb-core/user-config)
  (cb-project/user-config)
  (cb-ledger/user-config)
  (cb-bootstrap/disable-debugging))
```
Also, it's important that the file name matches the name of the feature a file provides (which is what `(provide 'personal-config)` sets up). If they do not match, the `require` mechanism will not work correctly.
---
Let me know how you go.

**Comment by biocyberman** [comment]
@chrisbarrett :+1: It works now. There is one thing however: changes in `personal-config.el` are not applied to the current Emacs instance after `SPC f e R`. So,  when I make changes there, I have to `eval` the elisp region or restart Emacs.

**Comment by chrisbarrett** [comment]
Yep. `require` won't load the same thing twice. When you make changes in your personal config, just run `M-x eval-buffer` as you go.
